packages/thirdweb/src/react/web/ui/prebuilt/NFT/index.tsx

```diff
@@ -0,0 +1,83 @@
+"use client";
+
+import { createContext, useContext } from "react";
+import type { ThirdwebContract } from "../../../../../contract/contract.js";
+import { tokenURI } from "../../../../../extensions/erc721/__generated__/IERC721A/read/tokenURI.js";
+import { uri } from "../../../../../extensions/erc1155/__generated__/IERC1155/read/uri.js";
+import { useReadContract } from "../../../../../react/core/hooks/contract/useReadContract.js";
+import type { NFTMetadata } from "../../../../../utils/nft/parseNft.js";
+import { MediaRenderer } from "../../MediaRenderer/MediaRenderer.js";
+import type { MediaRendererProps } from "../../MediaRenderer/types.js";
+
+export type NFTProviderProps = {
+  contract: ThirdwebContract;
+  tokenId: bigint;
+};
+
+export const NFTProviderContext = /* @__PURE__ */ createContext<
+  NFTProviderProps | undefined
+>(undefined);
+
+export function NFT(props: React.PropsWithChildren<NFTProviderProps>) {
+  return (
+    <NFTProviderContext.Provider value={props}>
+      {props.children}
+    </NFTProviderContext.Provider>
+  );
+}
+
+/**
+ * @internal
+ */
+function useNFTContext() {
+  const ctx = useContext(NFTProviderContext);
+  return ctx;
+}
+
+export type NFTMediaProps = MediaRendererProps & NFTProviderProps;
+
+NFT.Media = (props: NFTProviderProps) => {
+  const context = useNFTContext();
+  const contract = props.contract || context;
+  const mediaQuery = useReadContract(getNFTMedia, {
+    contract,
+    tokenId: props.tokenId,
+  });
```

Comment:
probably needs to be a custom `useQuery` because you want [useSuspenseQuery](https://tanstack.com/query/latest/docs/framework/react/reference/useSuspenseQuery) here

meaning it is up to the *consumer* of this component to wrap it in a `<Suspense>` boundary and display a loading state

[suspense docs](https://react.dev/reference/react/Suspense)